File: best-time-to-buy-and-sell-stock-ii/best-time-to-buy-and-sell-stock-ii.java
class Solution {
    public int maxProfit(int[] prices) {
        
        int max = 0;
        for(int i = 1;i < prices.length;i++){
            int currentPrice = prices[i-1];
            int nextDayPrice = prices[i];
            if(nextDayPrice > currentPrice) max += nextDayPrice - currentPrice; 
        }
        
        return max;

    }
}

File: best-time-to-buy-and-sell-stock-ii/best-time-to-buy-and-sell-stock-ii.kt
class Solution {
    fun maxProfit(prices: IntArray): Int {
        var max = 0
        for(i in 1 until prices.size){
            val current = prices[i]
            val prev = prices[i-1]
            if(current > prev) max += (current - prev)
        }
        return max
    }
}